### ISSUE TITLE
Use of metaclass' cause invalid errors 

### ISSUE THREAD

**Issue by plafosse** [microsoft/pyright#2000]
The following code using a metaclass results in an invalid error on the final line:

```python
class Foo(type):
    def __getitem__(cls:'Foo', item:int) -> int:
        return item + 1

class Bar(metaclass=Foo):
    pass

print(Bar[1])
```

The error given is:

```
Expected class type but received "Literal[1]"
```

VS Code extension running the latest 1.1.150

**Comment by erictraut** [comment]
Thanks for the bug report. I've added support for metaclasses that define `__getitem__`, `__setitem__`, and `__delitem__`.

**Comment by erictraut** [comment]
This is included in pyright 1.1.151, which I just published. It will also be included in the next release of pylance.
